pkg/operators/ebpf/attach.go

```diff
@@ -145,6 +153,88 @@ func (i *ebpfInstance) attachProgram(gadgetCtx operators.GadgetContext, p *ebpf.
 		return link.AttachLSM(link.LSMOptions{
 			Program: prog,
 		})
+	case ebpf.PerfEvent:
+		perfType := uint32(unix.PERF_TYPE_SOFTWARE)
+		perfConfig := uint64(unix.PERF_COUNT_SW_CPU_CLOCK)
+		perfSampleType := uint64(unix.PERF_SAMPLE_RAW)
+		frequency := uint64(0)
+		name, ok := strings.CutPrefix(p.SectionName, perfEventPrefix)
+		if !ok {
+			return nil, fmt.Errorf("perf_event programs require a name")
+		}
+
+		if tmp := i.config.GetString("programs." + name + ".perf.type"); tmp != "" {
+			switch tmp {
+			case "software":
+				perfType = unix.PERF_TYPE_SOFTWARE
+			default:
+				return nil, fmt.Errorf("unsupported perf.type %q", tmp)
+			}
+		} else {
+			return nil, fmt.Errorf("perf.type not specified for program %q", name)
+		}
```

Comment:
nit: can you consolidate these two on the same switch case?

(same in all places below)

```
case "":
    return nil, fmt.Errorf("perf.type not specified for program %q", name)
```